scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5240 @@
+    {X.variance(), X.standard_deviation()}
+    >>> X.skewness(), X.kurtosis()
+    {X.skewness(), X.kurtosis()}
+    >>> np.allclose(X.moment(order=6, kind='standardized'),
+    ...             X.moment(order=6, kind='central') / X.variance()**3)
+    True
+    >>> np.allclose(np.exp(X.logentropy()), X.entropy())
+    True
+
+    Pseudo-random samples can be drawn from
+    the underlying distribution using ``sample``.
+
+    >>> X.rng = np.random.default_rng(2354873452)
+    >>> X.sample(shape=(4,))
+    {repr(X.sample(shape=(4,)))}
+    """
+    # remove the indentation due to use of block quote within function;
+    # eliminate blank first line
+    example = "\n".join([line.lstrip() for line in example.split("\n")][1:])
+    return example
+
+
+class ContinuousDistribution:
+    r""" Class that represents a continuous statistical distribution.
+
+    Instances of the class represent a random variable.
+
+    Parameters
+    ----------
+    tol : positive float, optional
+        The desired relative tolerance of calculations. Left unspecified,
+        calculations may be faster; when provided, calculations may be
+        more likely to meet the desired accuracy.
+    validation_policy : {None, "skip_all"}
+        Specifies the level of input validation to perform. Left unspecified,
+        input validation is performed to ensure appropriate behavior in edge
+        case (e.g. parameters out of domain, argument outside of distribution
+        support, etc.) and improve consistency of output dtype, shape, etc.
+        Pass ``'skip_all'`` to avoid the computational overhead of these
+        checks when rough edges are acceptable.
+    cache_policy : {None, "no_cache"}
+        Specifies the extent to which intermediate results are cached. Left
+        unspecified, intermediate results of some calculations (e.g. distribution
+        support, moments, etc.) are cached to improve performance of future
+        calculations. Pass ``'no_cache'`` to reduce memory reserved by the class
+        instance.
+    rng : numpy.random.Generator
+        Random number generator to be used by any methods that require
+        pseudo-random numbers (e.g. `sample`).
+
+    Attributes
+    ----------
+    All parameters are available as attributes.
+
+    Methods
+    -------
+    support
+
+    plot
+
+    sample
+
+    moment
+
+    mean
+    median
+    mode
+
+    variance
+    standard_deviation
+
+    skewness
+    kurtosis
+
+    pdf
+    logpdf
+
+    cdf
+    icdf
+    ccdf
+    iccdf
+
+    logcdf
+    ilogcdf
+    logccdf
+    ilogccdf
+
+    entropy
+    logentropy
+
+    Notes
+    -----
+    The following abbreviations are used throughout the documentation.
+
+    - PDF: probability density function
+    - CDF: cumulative distribution function
+    - CCDF: complementary CDF
+    - entropy: differential entropy
+    - log-*F*: logarithm of *F* (e.g. log-CDF)
+    - inverse *F*: inverse function of *F* (e.g. inverse CDF)
```

Comment:
For most (currently implemented) continuous distributions it would indeed be correct.
But for all discrete distributions I believe that it won't be, because their CDF's are, by definition, not continuous. 

To illustrate:

```pycon
>>> from scipy.stats.distributions import bernoulli
>>> coin = bernoulli(1 / 2)
>>> quantiles = [0, 1/4, 1/2, 3/4, 1]
>>> coin.ppf(quantiles)
array([-1.,  0.,  0.,  1.,  1.])
>>> coin.cdf(coin.ppf(quantiles))
array([0. , 0.5, 0.5, 1. , 1. ])
```

This is in contradiction with the definition of the inverse, i.e. $f^{-1}(f(x)) = x$, so this isn't the inverse of the CDF.

There's also the chance that the term "inverse" gets confused with the "complement", i.e. the survival function in this case, since `sf` was also changed (which I like btw).

To conclude: `qf` is also twice as short as `icdf`, which for python users doesn't really matter, but if we also want to attract R users with this, then it really does matter a lot 😛.

Reply:
I think „inverse cdf“ is a very precise name. In general, quantiles are set-valued functionals of a distribution, set-valued because a cdf might have constant regions.
Here, we don’t want set-valued objects but the result of `numpy.quantile(x, method="inverted_cdf")`, i.e. $\mathrm{inf}\{x: p \leq cdf(x)\}$.

Reply:
> In general, quantiles are set-valued functionals of a distribution, set-valued because a cdf might have constant regions. Here, we don’t want set-valued objects but the result of `numpy.quantile(x, method="inverted_cdf")`, i.e. inf x : p ≤ c d f ( x ) .

This is indeed a correct description of the quantile function. But an [inverse function](https://mathworld.wolfram.com/InverseFunction.html) is something more specific, which additionally requires that $F^{-1}(F(x)) == x$ for all $x$ in the domain.
But this doesn't hold for the quantile function of e.g. discrete distributions, which proofs that the quantile function isn't the same as the inverse CDF.

Reply:
I think it's fine to be potentially imprecise and refer to this as an inverse cdf. When we get to the point where we are adding discrete and mixed distributions, we can add clarifying comments. I think `icdf` will be immediately understood by a greater proportion of users than `qf`, and I don't anticipate there being much confusion in the case of discrete and mixed distributions, since there is already long precedent for using inverse cdf in statistical software.